broadinstitute/pearl
# [JN-1299] refine at translation script

#### DESCRIPTION (include screenshots, and mobile screenshots for participant UX)

I was overzealous to think that proxy would be able to happen here - AT actually requires >1 governed user per proxy so my initial plan wouldn't work. 

This PR cleans up survey / profile data to the point that, from my initial testing, it seems good, just needs some much deeper QA than I have done. AFAIK the only major thing left is proxy account creation.

#### TO TEST:  *(simple manual steps for confirming core behavior -- used for pre-release checks)*
If you want to test, then you can:

Download data & atcp portal:
https://drive.google.com/file/d/1b7x-YzmRVwH3yO6GFkZoMjk-tvH07Dcy/view?usp=drive_link, https://drive.google.com/file/d/1aIsCiWLuxdOUU6fnpM5LM2Vc4X8OLppO/view?usp=drive_link

- Import portal
- Extract data zip file
- cd into the data zip file contents
- translate:
```bash
python3 ~/path/to/juniper/pepper-import/scripts/translate.py -DDj juniper_data_dict.xlsx -DDd pepper_data_dict.xlsx -I pepper_data.tsv -O juniper_data.csv -T translation_overrides.csv --limit 50
```
- delete admin notification in atcp portal
- import the `juniper_data.csv` file created by the translation script
- ensure that there's no FAILED participants
- click into the top couple participants and ensure there are no parse errors or anything
- verify profiles are properly filled out
- run `select * from answer where survey_stable_id = 'PREQUAL';` and verify pre-enroll answers are saved
   - NOTE: right now pre-enrollment responses in the UI are being displayed from the json blob even though refactors have made it to where pre-enrollment questions are actually saved as full surveyresponse objects. since this import infra only saves them as surveyresponses and doesn't fill in the json blob, these questions don't show up in the UI. IMO I think the UI should be refactored to use the surveyresponse object since (eventually) we'll want pre-enrolls to be full surveys anyway.
